### PR TITLE
scaffolder: Allow user or org, remove hard coded isOrg

### PR DESCRIPTION
**Pull request by timja**
## Hey, I just made a Pull Request!

Previously the code was hardcoded to organisations only, now you can also store in user repos.
Useful if you don't want to store the repo in org account, i.e. prototyping, learning backstage etc.

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [x] Regression tests added for bug fixes

### CODE REVIEW

File: docs/features/techdocs/creating-and-publishing.md [hunk 1]
```diff
@@ -30,11 +30,6 @@ the documentation template.
 Create an entity from the documentation template and you will get the needed
 setup for free.
 
-!!! warning Currently the Backstage Software Templates are limited to create
```
Comment (timja) [inline docs/features/techdocs/creating-and-publishing.md:33]: There's also a reference in https://github.com/spotify/backstage/blob/master/microsite/blog/2020-08-05-announcing-backstage-software-templates.md not sure if you want a blog post updated or not?

File: docs/features/techdocs/creating-and-publishing.md [hunk 2]
```diff
@@ -30,11 +30,6 @@ the documentation template.
...
-repositories inside GitHub organizations. You also need to generate an personal
```
Comment (timja) [inline docs/features/techdocs/creating-and-publishing.md:34]: do you want the rest of the warning maintained?

**Review by Rugvip — APPROVED** [review]
Awesome stuff, thanks! :grin:

File: docs/features/techdocs/creating-and-publishing.md (hunk 1 above)
Comment (Rugvip) [reply docs/features/techdocs/creating-and-publishing.md:33]: I'd say it'd be fine to leave the blog post intact. It's a bit of a snapshot in time, but it also doesn't make clear that you can't scaffold repos on user accounts tbh.

**Review by stefanalund — APPROVED** [review]

**Review by emmaindal — APPROVED** [review]
NIce! Thanks!

File: docs/features/techdocs/creating-and-publishing.md (hunk 2 above)
Comment (emmaindal) [reply docs/features/techdocs/creating-and-publishing.md:34]: @timja I think its fine to delete the entire warning for now! Thanks !

**Comment by freben** [comment]
@benjdlambert will you give it a look and then merge if you're happy about it?

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**